base_layer/transaction_components/src/key_manager/inner.rs

```diff
@@ -503,9 +529,68 @@ where TBackend: TransactionKeyManagerBackend + 'static
                     },
                 }
             },
+            TariKeyId::DHCommitmentMask {
+                public_key,
+                private_key,
+            } => {
+                let key = TariKeyId::from_str(private_key.to_string().as_str())
+                    .map_err(|_| KeyManagerServiceError::KeySerializationError)?;
+                let shared_secret = Box::pin(self.get_diffie_hellman_shared_secret(&key, public_key)).await?;
+                let commitment_mask_private_key = shared_secret_to_output_spending_key(&shared_secret)?;
+                Ok(commitment_mask_private_key)
+            },
+            TariKeyId::DHEncryptedData {
+                public_key,
+                private_key,
+            } => {
+                let key = TariKeyId::from_str(private_key.to_string().as_str())
+                    .map_err(|_| KeyManagerServiceError::KeySerializationError)?;
+                let shared_secret = Box::pin(self.get_diffie_hellman_shared_secret(&key, public_key)).await?;
+                let commitment_mask_private_key = shared_secret_to_output_encryption_key(&shared_secret)?;
+                Ok(commitment_mask_private_key)
+            },
+            TariKeyId::Encrypted { encrypted, key } => {
+                let key = TariKeyId::from_str(key.to_string().as_str())
+                    .map_err(|_| KeyManagerServiceError::KeySerializationError)?;
+                let private_key = Box::pin(self.decrypt_encrypted_key(encrypted, key)).await?;
+                Ok(private_key)
+            },
         }
     }
 
+    async fn created_encrypted_key(
```

Comment:
See comments regarding `pub async fn import_key(`
```suggestion
    pub async fn created_encrypted_key(
```